app/models/user.rb

```diff
@@ -32,6 +33,21 @@ def self.digest(string)
   def self.new_token
     SecureRandom.urlsafe_base64
   end
+  def self.from_omniauth(access_token)
+    data = access_token.info
+    user = access_token.uid.nil? ? User.where(uid: access_token.uid).first : User.where(email: data["email"]).first
+
+    # Uncomment the section below if you want users to be created if they don't exist
+    unless user
+      data["email"] = "#{access_token.uid}@facebook.com" if data["email"].nil?
```

Comment:
cần phải tính trường hợp log in từ facebook và user vs cùng 1 email.
kHi đó vẫn lưu đầy đủ thông tin nhưng chỉ là của 1 ngươi